services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbAsyncTable.java

```diff
@@ -667,4 +667,27 @@
     default CompletableFuture<T> updateItem(T item) {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Deletes a table in DynamoDb with the name and schema already defined for this DynamoDbTable.
+     * <p>
+     * Use {@link DynamoDbEnhancedClient#table(String, TableSchema)} to define the mapped table resource.
+     * <p>
+     * This operation calls the low-level DynamoDB API DeleteTable operation. Note that this is an asynchronous operation and that
+     * the table may not immediately deleted. You can use
+     * {@link software.amazon.awssdk.services.dynamodb.waiters.DynamoDbAsyncWaiter#waitUntilTableNotExists} in the underlying client.
+     * <p>
+     * Example:
+     * <pre>
+     * {@code
+     *
+     * mappedTable.deleteTable().join();
+     * }
+     * </pre>
+     *
+     * @return a {@link CompletableFuture} of {@link Void}.
+     */
+    default CompletableFuture<Void> deleteTable() {
```

Comment:
Should we return the `DeleteTableResponse`? Same as sync. I think we have feature requests on some existing APIs that are void.

Reply:
+1 to returning a response, even if it's empty.

Reply:
I had this argument a lot with put-item and create-table to try to add a return type, but we settled on not returning something. I still don't agree with that decision, but it'd be inconsistent to return something from this one.

Reply:
(I'm not saying we shouldn't return something here, just that we'd need a good reason to be inconsistent with the other APIs, or a plan to make the other APIs consistent with this one.)

Reply:
Oh, I completely forgot we have createTable already in this class that does not have a return type. In this case, I'd vote for consistency

Reply:
Had taken createTable as a reference for deleteTable , thus returning void.
I too will prefer keeping deleteTable consistent with createTable return type.